lock-api-objects/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -110,4 +110,8 @@ default List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTr
     void tryUnlock(Set<LockToken> tokens);
 
     long currentTimeMillis();
+
+    LockNamedTimestampResponse lockNamedTimestamp(String timestampName, int numFreshTimestamps);
+
+    long getSmallestLockedNamedTimestamp(String timestampName);
```

Comment:
Not sure if the "smallest" big adds anything here, especially since the implementation can move the value as well. I'd just say getNamedTimestamp? Or getLockedNamedTimestamp

Reply:
Yeah

Reply:
I am still not sure how to name these here. Most of the server bits use "named minimum timestamp lease" instead of talking about locks

I think I use lock because we already had immutable timestamp lock here (but, this has historically led people stray - at least in other contexts where teams think lock means it can never be unlocked if i don't unlock it)

Reply:
If I use the "lease" language I would do something like
```
NamedMinimumTimestampLeaseResponse acquireNamedMinimumTimestampLease(String, int);

long getSmallestLeasedNamedTimestamp();
```

Reply:
If you need the "Smallest" bit, I'd go for Min or something

Reply:
I like the "lease" vs lock, although it feels like a technicality: anyone that knows anything about distributed locks, needs to assume that they will expire if not refreshed. Otherwise you loose availability